### PR TITLE
Fix C and JS wrappers 

### PR DESCRIPTION
**Pull request by altafan**
* C wrappers: fixes use of `memcpy` and fixes creating/parsing/serializing a pedersen commit . Previosuly, the raw commitment was incorrectly truncated to 33 bytes while it actually is 64 bytes.
* JS wrappers: fixes the according pedersen wrappers of those cited above in order to allocate the correct size for the outputs and fixes checks on length of rangeproof function arguments.

### CODE REVIEW

File: lib/pedersen/index.js [hunk 1]
```diff
@@ -68,12 +68,12 @@ function commit(blindFactor, value, generator) {
  *  @summary Serializes a pedersen commitment.
  *  @return {Array} 33-bytes serialized pedersen commitment.
```
Comment (tiero) [inline lib/pedersen/index.js]: 64

Comment (tiero) [reply lib/pedersen/index.js]: How it is that input has 64 and output 33?

File: src/main.c [hunk 2]
```diff
@@ -182,8 +182,8 @@ int surjectionproof_parse(size_t *n_inputs, unsigned char *used_inputs, unsigned
   secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
   secp256k1_surjectionproof proof;
   memcpy(&(proof.n_inputs), n_inputs, sizeof(proof.n_inputs));
-  memcpy(proof.used_inputs, used_inputs, 32);
-  memcpy(proof.data, data, 8224);
+  memcpy(&(proof.used_inputs), used_inputs, 32);
+  memcpy(&(proof.data), data, 8224);
```
Comment (tiero) [inline src/main.c]: I don't get this magic bumber 🤔

Comment (altafan) [reply src/main.c]: https://github.com/ElementsProject/secp256k1-zkp/blob/secp256k1-zkp/include/secp256k1_surjectionproof.h#L50
those are the fixed length of the surjection proof fields.

File: lib/pedersen/index.js (hunk 1 above)
Comment (altafan) [reply lib/pedersen/index.js]: The commitment serialization takes a 64 byte raw pedersen commitment as input and return a 33 byte serialized commit as output.

**Comment by altafan** [comment]
@tiero

**Pull request merged**